src/mirrors.ts

```diff
@@ -0,0 +1,36 @@
+import { ConfigurationTarget, WorkspaceConfiguration } from "vscode";
+import fs from "fs";
+import path from "path";
+import { getConfigValue, metalsDir } from "./util";
+
+const mirrorProperty = "coursierMirror";
+
+export function getMirrorPath(
+  config: WorkspaceConfiguration
+): string | undefined {
+  const mirrorConfig = getConfigValue<string>(config, mirrorProperty);
+
+  const value = mirrorConfig?.value.trim();
```

Comment:
I would propose:
```typescript
/**
 * @returns path to the mirror config file described in https://get-coursier.io/blog/#mirrors
 */
export function getCoursierMirrorPath(
  config: WorkspaceConfiguration
): string | undefined {
  const mirrorConfig = getConfigValue<string>(config, mirrorProperty);

  const value = mirrorConfig?.value.trim();
  if (mirrorConfig && value && value.length > 0) {
    return writeMirrorFile(value, mirrorConfig.target);
  }
}

function writeMirrorFile(mirrorString: string, target: ConfigurationTarget) {
  const dotMetalsDir = metalsDir(target);
  const file = path.join(dotMetalsDir, "mirror.properties");
  const mirrorContents = [
    "metals.from=https://repo1.maven.org/maven2",
    `metals.to=${mirrorString}`,
  ].join("\n");
  fs.writeFileSync(file, mirrorContents);
  return file;
}
```
There is no any gain in checking if file already exists imo

Reply:
Changed!